lib/child-writer/index.js

```diff
@@ -170,14 +182,83 @@ exports.write = (image, drive, options) => {
     }
   }
 
+  // Timeout function that only runs if we exceed the specified timeout,
+  // terminating the server.
+  let terminationTimeout = null
+
+  // If the child process is created successfully it will be stored here.
+  let maybeChild = null
+
+  /**
+   * @summary Finish the child writer
+   * @function
+   * @private
+   *
+   * @param {Number} code - exit code
+   * @returns {Void}
+   *
+   * @example
+   * finish(EXIT_CODES.SUCCESS)
+   */
+  const finish = (code) => {
+    terminateServer()
+    clearTimeout(terminationTimeout)
+
+    if (code === EXIT_CODES.CANCELLED) {
+      return emitter.emit('done', {
+        cancelled: true
+      })
+    }
+
+    // We shouldn't emit the `done` event manually here
+    // since the writer process will take care of it.
+    if (code === EXIT_CODES.SUCCESS || code === EXIT_CODES.VALIDATION_ERROR) {
+      return null
+    }
+
+    return emitError(new Error(`Child process exited with error code: ${code}`))
+  }
+
+  /**
+   * @summary Reset the timeout countdown
+   * @function
+   * @private
+   *
+   * @param {ChildProcess} child - child process
+   *
+   * @example
+   * resetTerminationTimeout()
+   */
+  const resetTerminationTimeout = (child) => {
+    if (child) {
+      maybeChild = child
+    }
+
+    clearTimeout(terminationTimeout)
+    terminationTimeout = setTimeout(() => {
+      emitter.emit('timeout', {
+        timeout: CHILD_WRITER_TIMEOUT_MS
```

Comment:
What do you think about making this an `Error` object instead? For example:

```js
var error = new Error( 'Write process timed out' )
error.timeout = CHILD_WRITER_TIMEOUT_MS

emitter.emit('timeout', error)
```

Reply:
Done!